### PR TITLE
Chore: Add benchmark and improve docs for exception-ignore plugin

### PR DESCRIPTION
**Pull request by lujiajing1126**
Signed-off-by: Megrez Lu <lujiajing1126@gmail.com>

### Add details about the performance footprint of `exception-ignore` plugin
- [x] Add a benchmark for the improvement. Not applicable.
- [x] The benchmark result.
```text
/**
 * # JMH version: 1.33
 * # VM version: JDK 1.8.0_292, OpenJDK 64-Bit Server VM, 25.292-b10
 * # VM invoker: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/bin/java
 * # VM options: -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=54972:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8
 * # Blackhole mode: full + dont-inline hint (default, use -Djmh.blackhole.autoDetect=true to auto-detect)
 * # Warmup: 5 iterations, 10 s each
 * # Measurement: 5 iterations, 10 s each
 * # Timeout: 10 min per iteration
 * # Threads: 1 thread, will synchronize iterations
 * # Benchmark mode: Average time, time/op
 * <p/>
 * Benchmark                                             Mode  Cnt   Score   Error  Units
 * HierarchyMatchExceptionBenchmark.depthOneBenchmark    avgt   25  31.050 ± 0.731  ns/op
 * HierarchyMatchExceptionBenchmark.depthTwoBenchmark    avgt   25  64.918 ± 2.537  ns/op
 * HierarchyMatchExceptionBenchmark.depthThreeBenchmark  avgt   25  89.645 ± 2.556  ns/op
 */
```

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: docs/en/setup/service-agent/java-agent/How-to-tolerate-exceptions.md [hunk 1]
```diff
@@ -107,8 +107,15 @@ If an exception has the `@IgnoredException` annotation, the exception wouldn't b
...
 ## Recursive check
-Due to the wrapper nature of Java exceptions, sometimes users need recursive checking. Skywalking also supports it. Typically, we don't recommend setting this more than 10, which could cause a performance issue. Negative value and 0 would be ignored, which means all exceptions would make the span tagged in error status.
+Due to the wrapper nature of Java exceptions, sometimes users need recursive checking. Skywalking also supports it. 
 
 ```
-    statuscheck.max_recursive_depth=${SW_STATUSCHECK_MAX_RECURSIVE_DEPTH:1}
+statuscheck.max_recursive_depth=${SW_STATUSCHECK_MAX_RECURSIVE_DEPTH:1}
 ```
+
+According to the benchmark result, the exception check time is nearly proportional to the recursive depth being set.
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/How-to-tolerate-exceptions.md:116]: Please attach your benchmark result here.

**Comment by wu-sheng** [comment]
Please update `changes.md` in the root.

File: docs/en/setup/service-agent/java-agent/How-to-tolerate-exceptions.md (hunk 1 above)
Comment (lujiajing1126) [reply docs/en/setup/service-agent/java-agent/How-to-tolerate-exceptions.md]: Done

**Comment by lujiajing1126** [comment]
> Please update `changes.md` in the root.

Updated

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**